File: qc-core/src/main/java/com/eager/questioncloud/core/domain/hub/board/model/PostComment.java
package com.eager.questioncloud.core.domain.hub.board.model;

import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;

@Getter
public class PostComment {
    private Long id;
    private Long boardId;
    private Long writerId;
    private String comment;
    private LocalDateTime createdAt;

    @Builder
    public PostComment(Long id, Long boardId, Long writerId, String comment, LocalDateTime createdAt) {
        this.id = id;
        this.boardId = boardId;
        this.writerId = writerId;
        this.comment = comment;
        this.createdAt = createdAt;
    }

    public static PostComment create(Long boardId, Long writerId, String comment) {
        return PostComment.builder()
            .boardId(boardId)
            .writerId(writerId)
            .comment(comment)
            .createdAt(LocalDateTime.now())
            .build();
    }

    public void modify(String comment) {
        this.comment = comment;
    }
}

File: qc-core/src/main/kotlin/com/eager/questioncloud/core/domain/post/model/PostComment.kt
package com.eager.questioncloud.core.domain.post.model

import java.time.LocalDateTime

class PostComment(
    val id: Long = 0,
    val postId: Long,
    val writerId: Long,
    var comment: String,
    val createdAt: LocalDateTime = LocalDateTime.now(),
) {
    fun modify(comment: String) {
        this.comment = comment
    }

    companion object {
        fun create(postId: Long, writerId: Long, comment: String): PostComment {
            return PostComment(postId = postId, writerId = writerId, comment = comment)
        }
    }
}
